elasticdl/python/master/rendezvous_server.py

```diff
@@ -68,9 +75,21 @@ def get_rendezvous_port(self):
         return self._rendezvous_port
 
     def get_worker_host_rank(self, host):
+        if self._next_worker_hosts and self._rendezvous_completed:
+            self._init_rendezvous_server()
+
         # -1 if host not in worker_hosts list.
         if host not in self._worker_hosts:
             return -1
+
+        if not self._rendezvous_completed:
+            self._ready_worker_hosts.add(host)
+            # If all active workers in the rendezvous are ready,
+            # the server can start to set hosts for the next rendezvous
```

Comment:
After a worker gets its rank, it needs to call `hvd.init()`. If the master set a new host plan before worker calls  `hvd.init()`, there may be crash/hang.

So the master needs to wait a few seconds after the last worker gets the rank to start the next round of rendezvous.